services/web-server/src/graphql/WorkerManager.graphql

```diff
@@ -1,9 +1,12 @@
 type WorkerManagerWorkerPoolSummary {
     workerPoolId: String!
+    workerPoolId1: String
+    workerPoolId2: String
```

Comment:
> by the time it makes it to a GraphQL request it's formatted as just a regular old workerPoolId

that is the case here

Reply:
I'm confused then -- why are there `workerPoolId1/2` properties in the graphql?

Reply:
oh, I see why you have this question.

So those two fields are added to the type definition because I needed to split the Id in the loader (see this commit https://github.com/taskcluster/taskcluster/pull/865/commits/b1660387e407e5a32f2a8877f3d528db89584320 ). So that's the opposite operation of what you were asking about. (I massage the data in the loader for state/props management convenience and code simplification. This can easily be changed once we deprecate the whole `/` thing)

Reply:
> I'm confused then -- why are there workerPoolId1/2 properties in the graphql?

See above. Those two props that caught your eye are populated/present only when the data comes from the loader.

Admittedly, that operation can also be made during the component state initiation (the split will have to be done twice for each half - I thought it would be silly, so I opted for splitting it once in the loader)

Reply:
That latter bit is what I'm suggesting - handle the splitting and joining in the component, rather than in web-server, so that the split isn't visible in the GraphQL.

Reply:
I need to check that, because I am sure I had reasons for doing this (unfortunately, I don't remember now what they were, as this was written on Monday)

Reply:
so the change you're proposing will result in this sort of code in the state initialization:

```javascript
...
  workerPoolId1: this.props.allowEditWorkerPoolId 
      ? this.props.workerPool.workerPoolId1 
      : splitWorkerPoolId(this.props.workerPool.workerPoolId).provisionerId,
...
```

I am not super euphoric about code like this, but if we are happy, I can make the change. Why is it important to keep it out of the web-server/graphql code tho?

Reply:
That code snippet shows one way to accomplish this, but I think some refactoring could simplify it further.  It's hard to tell without context, but that snippet does make me wonder why there's a `workerPoolId1` property on a `workerPool` object.

It's important because separation of the workerPoolId into two parts is a detail of how it's presented and edited in the UI.  Limiting the number of places where that separation limits the cognitive load for the reader -- for example, a reader of web-server could spend quite a while wondering why there are two different numbered "workerPoolId" properties, when a worker pool only has one ID.

Reply:
Yeah, that makes sense. Web-server is an essential part of the ui tho.... it's not even a server in the sense that it doesn't talk to db directly, it just relays requests and data back and forth.... do people get to read it without having to work on the ui as well?

>  It's hard to tell without context

The context is in the PR, isn't it?

> I think some refactoring could simplify it further.

Can you explain the refactoring in more detail? (did you mean the snippet itself or do you already see the code as complex, having read it thoughtfully?)

Reply:
I meant refactoring the snippet (and the related changes since it was prefaced with "this sort of").  What I have in mind is a component
```jsx
<WorkerPoolIdField
  label="Worker Pool ID"
  name="workerPoolId"
  required
  disabled={!this.props.allowEditWorkerPoolId}
  autofocus={this.props.allowEditWorkerPoolId}
  workerPoolId={this.props.workerPool.workerPoolId}
  onChange={this.handleInputChange}
/>
```
That component would handle all of the splitting and joining internally, and otherwise work much like `TextField` as far as `WMWorkerPoolEditor` is concerned.

Reply:
oops, missed the first two questions:

> do people get to read it without having to work on the ui as well?

sure, people can read whatever they'd like.  The fewer things they need to understand to read it, the better.  In this case, web-server is translating from GraphQL to REST.  Adding a second purpose "to split this identifier since the UI uses two text fields" increases the number of things to think about while working on that service.

> The context is in the PR, isn't it?

Not the context of the "sort of code" you described -- the few lines you included didn't give me a complete picture of what you intend.

Reply:
I chose a different implementation, and I explained my reasons. It seems like we are running in circles

Reply:
do people get to read it without having to work on the ui as well tho?

Reply:
> do people get to read it without having to work on the ui as well?

sure, people can read whatever they'd like. The fewer things they need to understand to read it, the better. In this case, web-server is translating from GraphQL to REST. Adding a second purpose "to split this identifier since the UI uses two text fields" increases the number of things to think about while working on that service.

Reply:
OK, fair enough, I'll withdraw my review.

Reply:
> OK, fair enough, I'll withdraw my review.

Not sure what you mean, but on the point of removing things from graphql - I'll make that change, I just wanted to understand the reason you requested it